src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Metadata/DefaultModelMetadataProvider.cs

```diff
@@ -84,6 +93,13 @@ private ModelMetadataCacheEntry CreateCacheEntry(ModelMetadataIdentity key)
             return new ModelMetadataCacheEntry(metadata, details);
         }
 
+        private ModelMetadata GetMetadataForObjectType()
+        {
+            var key = ModelMetadataIdentity.ForType(typeof(object));
+            var entry = CreateCacheEntry(key);
```

Comment:
`GetMetadataForProperties()` first gets `modelType`'s (the container's) metadata, duplicating most of `GetMetadataForType()`. My feedback is to maintain the consistency or just to use `GetMetadataForType()` for that part of `GetMetadataForProperties()`.